src/sentry/api/endpoints/organization_events.py

```diff
@@ -488,6 +494,8 @@ def _dashboards_data_fn(scoped_dataset, offset, limit, scoped_query, dashboard_w
                 )
 
                 if does_widget_have_split and not has_override_feature:
+                    split_dataset: ModuleType
+
```

Comment:
rather than `ModuleType` (which makes these `Any`) this should use a `Protocol` describing the shape of the function being called here

Reply:
actually yeah instead of passing modules around -- pass the `.query` method directly

and:

```python
class _QueryCallable(Protocl):
   def __call__(self, ...):  # left as an exercise to PR author: fill out the shared shape of `query` across the modules
```